rust/arrow/src/array/array_binary.rs

```diff
@@ -373,6 +376,45 @@ impl From<Vec<Vec<u8>>> for FixedSizeBinaryArray {
     }
 }
 
+impl From<Vec<Option<Vec<u8>>>> for FixedSizeBinaryArray {
+    fn from(data: Vec<Option<Vec<u8>>>) -> Self {
+        let len = data.len();
+        assert!(len > 0);
+        // try to estimate the size. This may not be possible no entry is valid => panic
```

Comment:
This behavior (panic'ing') doesn't seem ideal, though I realize there isn't much useful to do when converting a Vec of entirely `None` -- maybe we could just return a zero length array. 

Could definitely be done as a follow on PR

Reply:
Note that in general we should avoid using these because they require two allocations (rust's `Vec` and arrow buffers). This function is mostly useful for testing.

I would be ok with replacing them by the `FromIter` constructor, which is more performance, more general, and has the same ergonomics (`from(vec![].into_iter())` instead of `from(vec![...])` for a vector). This way we do not need to worry about these.

The challenge with fixed sized items is that they require knowledge of the size. This would be nicely solved by accepting `Option<[T; T: usize]>`, but Rust's support for constant generics is slim atm.

Reply:
Filed https://issues.apache.org/jira/browse/ARROW-10903